packages/adapter-next/src/hooks/snippet-read.ts

```diff
@@ -62,33 +62,48 @@ export const snippetRead: SnippetReadHook<PluginOptions> = async (
 			const repeat = data.model.config?.repeat ?? false;
 			const allowText = data.model.config?.allowText ?? false;
 
+			const allowVariants = Boolean(data.model.config?.variants);
+			const variant = (path: string, prefix: "" | "\n    " = "") =>
+				allowVariants ? `${prefix}className={${path}.variant}` : "";
+
+			const path = dotPath(fieldPath);
+
 			let code;
 			if (!repeat && !allowText) {
 				code = await format(
 					stripIndent`
-					<PrismicNextLink field={${dotPath(fieldPath)}}>Link</PrismicNextLink>
+					<PrismicNextLink field={${path}} ${variant(path)}>Link</PrismicNextLink>
 				`,
 					helpers,
 				);
 			} else if (!repeat && allowText) {
 				code = await format(
 					stripIndent`
-					<PrismicNextLink field={${dotPath(fieldPath)}} />
+					<PrismicNextLink field={${path}} ${variant(path)}/>
 				`,
 					helpers,
 				);
 			} else if (repeat && !allowText) {
```

Comment:
❓To prevent the problem you have, we could have duplicated even more the code no? Having PrismicnextLink for each use case, and so with or without the className.